MMM-RandomPhoto.js

```diff
@@ -70,8 +72,18 @@ Module.register("MMM-RandomPhoto",{
     },
 
     fetchImageList: function() {
+        var self = this;
+
+        Log.info(this.name + " fetching image list");
         if (typeof this.config.repositoryConfig.path !== "undefined" && this.config.repositoryConfig.path !== null) {
             this.sendSocketNotification('FETCH_IMAGE_LIST');
+
+            if (this.config.scanInterval != -1) {
```

Comment:
1. Conflict

Could you please explain this in more detail?  I'm not really sure I see how there's a conflict.  The two timers should be independent.

2. startPaused set to true

I think the issue is that this.load() is getting called on line 107 when startPaused is set to true?  Consequently even though startPaused is set to true, the image will keep cycling.  

So to fix that, the IMAGE_LIST notification needs to be disconnected from loading an image - at least after an image has already been loaded?

Reply:
1. Conflict

Well this also has to do with the fact that we are calling `this.load()` on every `IMAGE_LIST` notification received. In this case both timers are triggering the load of the next image. That's what I mean with "conflict".

2. startPaused set to true

Well `this.load()` is always called in the [`resumeImageLoading` function](https://github.com/MystaraTheGreat/MMM-RandomPhoto/blob/master/MMM-RandomPhoto.js#L100-L112). Regardless of what `startPaused` is set to.
[`startHidden` actually handles](https://github.com/MystaraTheGreat/MMM-RandomPhoto/blob/a4d1e57130056dd718b7b278331aa129c3ad4ac2/MMM-RandomPhoto.js#L411) the call when receiving the notification.

After giving it a bit of thought, it is probably best to just get rid of [lines 411 to 413](https://github.com/MystaraTheGreat/MMM-RandomPhoto/blob/a4d1e57130056dd718b7b278331aa129c3ad4ac2/MMM-RandomPhoto.js#L411-L413).

We are calling the `resumeImageLoading` function already during MM²'s extended build-in [`resume` function](https://github.com/MystaraTheGreat/MMM-RandomPhoto/blob/a4d1e57130056dd718b7b278331aa129c3ad4ac2/MMM-RandomPhoto.js#L421-L423). So that should get the image loading started in the first place.

But this needs to be tested to see if the app behaves like it should with all the different config settings - especially `startHidden` and `startPaused`.

Reply:
Okay.  It does look as if they would solve the initial issue.  But I agree that it's necessary to check the startHidden and startPaused.  I'll give it a go here and see what happens.

Reply:
Hm, no, doesn't seem to work.  The image loading never starts.
A messier fix might be to add a firstImageLoaded flag to indicate whether a first image has loaded or not - and not to call resumeImageLoading if firstImageLoaded is true.

But it seems inelegant.

Reply:
If we delete lines 411 to 413 we need to also adjust [these lines](https://github.com/MystaraTheGreat/MMM-RandomPhoto/blob/a4d1e57130056dd718b7b278331aa129c3ad4ac2/MMM-RandomPhoto.js#L366-L369).
But the problem actually still persists. No display is being shown, because the image loading just takes more time.
So actually invoking the image **showing** on the `IMAGE_LIST` socketNotification still makes sense (lines 411 to 413).

So I would agree the easiest solution probably is to introduce some kind of flag / parameter to indicate if it is an "initial image loading" process or an "update image loading" process.

Some thoughts without going deeper into details:
- maybe add an additional `UPDATE_IMAGE_LIST` socketNotification?
- somehow wait for `this.imageList`'s lenght to be `> 0`?
- maybe work with `async` and `await` to wait for some executions?

Reply:
I'm now testing a version that uses an UPDATE_IMAGE_LIST socket notification.  On reception, all it does it replace the imageList.  It doesn't do anything regarding resumption.

I haven't followed up on waiting for the imageList length to be >0 or using async/wait.  These changes don't seem to be necessary?

Reply:
The above version seems to work.  However, I've been having difficulty with the calendar not updating as well, so I'm going to take the simpler option of just restarting MM every day.

Feel free to use the push or not as you like :)